pytensor/tensor/math.py

```
@@ -107,6 +107,15 @@
     float64_atol = 1e-8
 
 
+def __getattr__(name):
+    if name == "MaxandArgmax":
+        warnings.warn(
+            "The class `MaxandArgmax` has been deprecated. "
+            "Call `Max` and `Argmax` seperately as an alternative.",
+            FutureWarning,
+        )
+
```

This has to raise some sort of error because `MaxAndArgmax` no longer exist. It can't simply be a warning. I think it's better to just remove it

---

should we raise an ```AssertionError``` here?

---

Whatever error would be raised if this helper was not here, `AttrubetError` sounds about right but confirm with trying without the special code

---

The helper also needs to raise the standard error for things other than `MaxandArgmax`

---

Confirmed that ```AttributeError``` is raised

---

And is it still raising for other things that don't exist and are not `MaxAndArgmax`?

---

This is how we implement it in pymc for reference: https://github.com/pymc-devs/pymc/blob/f44071bdda363f743548187d3e124a027adfdb77/pymc/distributions/transforms.py#L54-L67

Although according to the PEP it should probably have a `!r` in the standard AttributeError message:  https://peps.python.org/pep-0562/#rationale

---

In our case, we won't have to give a warning right?

---

Yes in our case we just raise a more informative `AttributeError` instead of the default one

---

In your last change you made this useless, it's the same as if we didn't implement `__getattr__` at all. The point of adding it is to have a custom message for the deprecated Op, but have the default behavior for everything else.

---

Rectified this